compiler/rustc_passes/src/errors.rs

```diff
@@ -1045,11 +1045,10 @@ pub(crate) struct AbiInvalidAttribute {
 }
 
 #[derive(Diagnostic)]
-#[diag(passes_unrecognized_field)]
-pub(crate) struct UnrecognizedField {
+#[diag(passes_unrecognized_argument)]
+pub(crate) struct UnrecognizedArgument {
     #[primary_span]
     pub span: Span,
-    pub name: Symbol,
```

Comment:
it's a little unfortunate that the symbol is removed from the error here now. In another case earlier I thought it didn't matter so much but having the name there can be nice during testing, i.e. the main error string will list the attribute that's an error. If there are multiple errors on one line, it's afaik hard to deduce which one was the error based just on the span.

Reply:
There are a few cases like this. E.g. for this code:
```
#[macro_export(not_local_inner_macros)]
```
The old error was:
```
error: `not_local_inner_macros` isn't a valid `#[macro_export]`
```
and the new error is:
```
invalid `#[macro_export]` argument
```
which doesn't mention the invalid argument, because that argument might be a non-identifer, e.g. a string literal.
It could be made to mention it, it's requires an extra case, which is a little more code and complexity, and didn't seem worth it for these quite obscure error messages.

Reply:
hmm, no I do agree the error is still perfectly clear for users and the value might be small